app/web/features/auth/password/SetNewPassword.tsx

```diff
@@ -0,0 +1,155 @@
+import { Empty } from "google-protobuf/google/protobuf/empty_pb";
+import stringOrFirstString from "utils/stringOrFirstString";
+import { useAuthContext } from "features/auth/AuthProvider";
+import { Container } from "@material-ui/core";
+import { Typography } from "@material-ui/core";
+import StyledLink from "components/StyledLink";
+import TextField from "components/TextField";
+import HtmlMeta from "components/HtmlMeta";
+import { useMutation } from "react-query";
+import { useForm } from "react-hook-form";
+import makeStyles from "utils/makeStyles";
+import { useRouter } from "next/router";
+import { AUTH } from "i18n/namespaces";
+import Button from "components/Button";
+import { useTranslation } from "i18n";
+import Alert from "components/Alert";
+import { loginRoute } from "routes";
+import { RpcError } from "grpc-web";
+import { service } from "service";
+
+const useStyles = makeStyles((theme) => ({
+  form: {
+    "& > * + *": {
+      marginBlockStart: theme.spacing(1),
+    },
+  },
+  standardContainer: {
+    marginTop: theme.spacing(2),
+    paddingLeft: theme.spacing(2),
+    paddingRight: theme.spacing(2),
+    paddingBottom: theme.spacing(2),
+    flex: 1,
+  },
+  main: {
+    padding: theme.spacing(0, 3),
+  },
+  textField: {
+    "& > div": {
+      width: "100%",
+      marginBottom: theme.spacing(2),
+      [theme.breakpoints.up("md")]: {
+        width: theme.typography.pxToRem(400),
+      },
+    },
+  },
+}));
+
+export default function SetNewPassword() {
+  const { authState } = useAuthContext();
+  const { t } = useTranslation(AUTH);
+  const formClass = useStyles();
+  const { handleSubmit, register } =
+    useForm<{ newPassword: string; newPasswordCheck: string }>();
+
+  const router = useRouter();
+  const resetToken = stringOrFirstString(router.query.token);
+  const isResetTokenOk =
+    !!resetToken && typeof resetToken === "string" && resetToken !== ""; // TODO: needed??
+
+  const { error, isLoading, isSuccess, mutate } = useMutation<
+    Empty,
+    RpcError,
+    string
+  >((newPassword) =>
+    service.account.CompletePasswordResetV2(resetToken as string, newPassword)
+  );
+
+  const onSubmit = handleSubmit(({ newPassword, newPasswordCheck }) => {
+    if (newPassword !== newPasswordCheck) {
+      alert(t("change_password_form.password_mismatch_error"));
+      return;
+    }
+
+    mutate(newPassword);
+  });
+
+  // TODO: needed?
+  if (authState.authenticated) {
```

Comment:
don´t let the user change it´s password if he is logged in to avoid weird scenarios/edge cases?